### PR TITLE
Parsing body.data into list of maps with columns as keys and values as native types

### PR DESCRIPTION
**Pull request by forest**

### CODE REVIEW

File: lib/avalanche/request.ex [hunk 1]
```diff
@@ -84,11 +82,17 @@ defmodule Avalanche.Request do
     request_id = get_request_id()
     params = [requestId: request_id]
 
-    options = [base_url: request.url, params: params, auth: {:bearer, request.token}]
+    options = [
+      base_url: request.url,
+      params: params,
+      auth: {:bearer, request.token},
+      retry: [delay: 100, max_retries: 3]
```
Comment (sarahdasko) [inline lib/avalanche/request.ex:89]: should we make this configurable?

File: test/integration_test.exs [hunk 2]
```diff
@@ -3,42 +3,79 @@ defmodule AvalancheIntegrationTest do
...
-      assert {:ok, %Avalanche.Response{} = response} = Avalanche.run("select ?;", [33], c.options)
-      assert response.status == 200
+      assert {:ok, %Avalanche.Result{} = result} = Avalanche.run("select ?;", [33], c.options)
+      assert result.num_rows == 1
+    end
+
+    test "parses result body into list of maps", c do
+      assert {:ok, %Avalanche.Result{} = result1} =
+               Avalanche.run(
+                 "SELECT *, 9 as number FROM SNOWFLAKE_SAMPLE_DATA.TPCH_SF1.ORDERS LIMIT ?",
```
Comment (sarahdasko) [inline test/integration_test.exs:46]: should we have a sort on this query to ensure that the row that you are looking for is the first one returned?

File: lib/avalanche/request.ex (hunk 1 above)
Comment (forest) [reply lib/avalanche/request.ex:89]: Yes, but not sure we will keep this. It only handles retries for 50x errors. So we will need other retry handlers for things like 429s.

File: test/integration_test.exs (hunk 2 above)
Comment (forest) [reply test/integration_test.exs:46]: good call.

**Review by aschreifels — APPROVED** [review]
Barring @sarahdasko comments I am good

**Pull request merged**